app/src/main/java/com/macbitsgoa/ard/utils/HandleFirebaseMessages.java

```diff
@@ -0,0 +1,66 @@
+package com.macbitsgoa.ard.utils;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.util.Log;
+
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Map;
+
+public class HandleFirebaseMessages extends FirebaseMessagingService {
+
+    public static Bitmap getBitmapFromURL(String src) {
+        try {
+            URL url = new URL(src);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setDoInput(true);
+            connection.connect();
+            InputStream input = connection.getInputStream();
+            return BitmapFactory.decodeStream(input);
+        } catch (IOException e) {
+            // Log exception
+            Log.e("TAG", "fucked man " + e.getMessage());
+            return null;
+        }
+    }
+
+    @Override
+    public void onMessageReceived(RemoteMessage remoteMessage) {
+
+        Map<String, String> data = remoteMessage.getData();
+
+        if (data != null) {
+
+            String type = data.get("type");
+            NotificationHelper helper = new NotificationHelper(this);
+            switch (type) {
+                case "picture":
+                    Bitmap image = getBitmapFromURL(data.get("imageURL"));
+                    helper.pictureNotif(image, data.get("title"));
```

Comment:
There are a few more values that you should include so that the sender has complete control over the notification
send the image at https://pasteboard.co/Gz5d1OB.png

also consider adding ->
link -> url or within app on clicking notif
id -> an id for this notification..in case the sender make a mistake and wants to update the notification